extensions/mssql/src/util/dataCache.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export class DataCache {
+
+	millisecondsToLive: number;
```

Comment:
It's usually better to have a TTL per cache element, rather than a TTL for the entire cache

Reply:
There is only a single element here.

Reply:
Would it make more sense to have a multi element cache? It seems weird to have this functionality without being able to store multiple KV pairs

Reply:
It wasn't really meant to be for a collection. It would be more for the case when you need to do something that doesn't change very often, like a http API call to get some daily statistics. The callback function will query the API but only once a day if you set the TTL to that. We wouldn't want to make a http round trip call every time to get the same exact data. Furthermore, the callback being passed in would be different for fetching a different item. I am not so that making this a collection would make sense in this case.

Reply:
That makes sense!